F841 fix since 0.0.264 can break code

```shellsession
$ curl -O https://raw.githubusercontent.com/home-assistant/core/03710e58b5cc3cd1e2472d2e497bf5780695eca5/tests/components/homekit_controller/test_init.py
$ ruff test_init.py
test_init.py:150:5: F841 [*] Local variable `is_connected` is assigned to but never used
test_init.py:218:5: F841 [*] Local variable `is_available` is assigned to but never used
Found 2 errors.
[*] 2 potentially fixable with the --fix option.
$ ruff --fix test_init.py
Found 6 errors (6 fixed, 0 remaining).
```

Note found 2 vs 6 errors. After the `--fix`:
```shellsession
$ python3 -m test_init
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 185, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/usr/lib/python3.8/runpy.py", line 155, in _get_module_details
    code = loader.get_code(mod_name)
  File "<frozen importlib._bootstrap_external>", line 981, in get_code
  File "<frozen importlib._bootstrap_external>", line 911, in source_to_code
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File ".../test_init.py", line 116
    nonlocal is_connected
    ^
SyntaxError: no binding for nonlocal 'is_connected' found
```

Haven't actually checked, but I suppose the fix for these issues is removing too many of the same named variables, not only the two it flagged errors for in non-fix mode.

--- a/crates/ruff/resources/test/fixtures/pyflakes/F841_0.py
+++ b/crates/ruff/resources/test/fixtures/pyflakes/F841_0.py
@@ -126,3 +126,22 @@ def f(x: int):
 def f():
     if any((key := (value := x)) for x in ["ok"]):
         print(key)
+
+
+def f() -> None:
+    is_connected = False
+
+    class Foo:
+        @property
+        def is_connected(self):
+            nonlocal is_connected
+            return is_connected
+
+        def do_thing(self):
+            # This should resolve to the `is_connected` in the function scope.
+            nonlocal is_connected
+            print(is_connected)
+
+    obj = Foo()
+    obj.do_thing()
+

--- a/crates/ruff/src/checkers/ast/mod.rs
+++ b/crates/ruff/src/checkers/ast/mod.rs
@@ -291,14 +291,14 @@ where
                     }
 
                     // Mark the binding in the defining scopes as used too. (Skip the global scope
-                    // and the current scope.)
+                    // and the current scope, and, per standard resolution rules, any class scopes.)
                     for (name, range) in names.iter().zip(ranges.iter()) {
                         let binding_id = self
                             .semantic_model
                             .scopes
                             .ancestors(self.semantic_model.scope_id)
                             .skip(1)
-                            .take_while(|scope| !scope.kind.is_module())
+                            .filter(|scope| !(scope.kind.is_module() || scope.kind.is_class()))
                             .find_map(|scope| scope.get(name.as_str()));
 
                         if let Some(binding_id) = binding_id {

